vmdb/app/models/provider_foreman.rb

```diff
@@ -34,4 +34,35 @@ def build_managers
     build_provisioning_manager unless provisioning_manager
     build_configuration_manager unless configuration_manager
   end
+
+  def self.process_tasks(options)
+    raise "No ids given to process_tasks" if options[:ids].blank?
+    if options[:task] == "refresh_ems"
+      refresh_ems(options[:ids])
+      create_audit_event(options)
+    else
+      options[:userid] ||= "system"
+      raise_exception(options)
+      invoke_tasks_queue(options)
+    end
+  end
+
+  def self.create_audit_event(options)
+    msg = "'#{options[:task]}' successfully initiated for #{ApplicationController.new.pluralize(options[:ids].length,
```

Comment:
models should not be calling into application controller....this should be a ui_lookup call I believe.